view/frontend/web/internals/template/autocomplete/products.js

```diff
@@ -15,7 +15,8 @@ define([], function () {
 
         getItemHtml: function ({item, components, html}) {
             return html`<a class="algoliasearch-autocomplete-hit" href="${item.__autocomplete_queryID != null ? item.urlForInsights : item.url}"
-                data-objectId=${item.objectID} data-index=${item.__autocomplete_indexName} data-queryId=${item.__autocomplete_queryID}>
+                data-objectId=${item.objectID} data-index=${item.__autocomplete_indexName} data-queryId=${item.__autocomplete_queryID}
+                data-position=${item.__autocomplete_id + 1}>
```

Comment:
This logic seems suspect to me and will not derive a proper position index. I think we should extract that position at the `transformResponse` callback where we can access the full hits array or better yet let's look at using `createAlgoliaInsightsPlugin`.